doc/sphinx/source/graph-basics.rst

```diff
@@ -1,24 +1,120 @@
-
+#############
+
+Overview
+========
+
+This section provides a brief overview of some concepts used in the nGraph 
+Library. It also introduces new ideas regarding our unique departure from the 
+first generation of deep learning software design. 
+
+The historical dominance of GPUs at the beginning of the current 
+:abbr:`DL (Deep Learning)` boom means that many framework authors made 
+GPU-specific design decisions at a very deep level. Those assumptions created 
+an "ecosystem" of frameworks that all behave essentially the same at the
+framework's hardware abstraction layer: 
+
+* The framework expects to own memory allocation. 
+* The framework expects the execution device to be a GPU. 
+* The framework expects complete control of the GPU, and that the device doesn't 
+  need to be shared. 
+* The framework expects that developers will write things in a `SIMT-friendly`_ 
+  manner, thus requring only a limited set of data layout conventions.    
+  
+Some of these design decisions have implications that do not translate well to 
+the newer or more demanding generation of **adaptable software**. For example, 
+most frameworks that expect full control of the GPU devices experience their 
+own per-device inefficiency for resource utilization whenever the system 
+encounters a bottleneck. 
+
+Most framework owners will tell you to refactor the model in order to remove the 
+unimplemented copy, rather than attempt to run multiple models in parallel, or 
+attempt to figure out how to build graphs more efficiently. In other words, if 
+a model requires any operation that hasn't been implemented on GPU, it must wait 
+for copies to propagate from the CPU to the GPU(s). An effect of this 
+inefficiency is that it slows down the system. Data scientists who are facing a 
+large curve of uncertainty in how large (or how small) the compute-power needs 
+of their model will be, investing heavily in frameworks reliant upon GPUs may 
+not be the best decision.  
+
+Meanwhile, the shift toward greater diversity in deep learning **hardware devices** 
+requires that these assumptions be revisited. Incorporating direct support for 
+all of the different hardware targets out there, each of which has its own 
+preferences when it comes to the above factors, is a very heavy burden 
+on framework owners.
+
+Adding the nGraph compiler to the system lightens that burden by raising the 
+abstraction level, and by letting any hardware-specific backends make these 
+decisions automatically. The nGraph Compiler is designed to be able to take into 
+account the needs of each target hardware platform, and to achieve maximum 
+performance.
+
+This makes things easier on framework owners, but also (as new models are developed) 
+on data scientists, who will not have to keep in mind nearly as many low-level 
+hardware details when architecting their models with layers of complexity for 
+anything other than a :abbr:`Just-in-Time (JIT)` compilation.     
```

Comment:
Might just replace with "hardware-specific layers of complexity". JIT per se is not a concern that should command the data scientist's attention